### ISSUE TITLE
About the GTA2Cityscapes images and SSL model

### ISSUE THREAD

**Issue by zhyx12** [liyunsheng13/BDL#18]
Thanks for sharing your code. I want to know the iteration number of provided GTA2Cityscapes dataset, SSL_step1 and SSL_step2 model, is K=1 or K=2? or after training, will we get mIoU of 47.2 or 48.5?

**Comment by liyunsheng13** [comment]
I'm sorry your question is not very clear to me. Could you provide more details such as which is the iteration number you mean?

**Comment by zhyx12** [comment]
The k here means the outer iteration number in Algorithm1. Also in Table 2, line 4 (k=1) get 47.2, and line 7 (k=2) get 48.5

**Comment by liyunsheng13** [comment]
What is the exact question you want to ask? I'm still  confused. What is the iterations number you refer to and what is the GTA2Cityscapes dataset you refere to?

**Comment by zhyx12** [comment]
Sorry for the uncleared description. GTA2Cityscapes dataset means the link [GTA5 as Cityscapes](https://drive.google.com/open?id=1OBvYVz2ND4ipdfnkhSaseT8yu2ru5n5l) you provided

**Comment by liyunsheng13** [comment]
The translated images are based on the model with mIoU 47.2

**Comment by zhyx12** [comment]
Thanks for your reply, as the translated images based on adaptation model with mIoU 47.2, is it means we just need to train adaptation model and don't need to train Style transfer model(CycleGAN) again to get the final result(48.5)?

**Comment by liyunsheng13** [comment]
Yes. I provide a simple way to reproduce the best result in my paper. But if you are interested in reproducing the whole training process, you can train CycleGAN with the code I uploaded.

**Comment by zhyx12** [comment]
Get it. Thank you so much for your attention and patience.

**Comment by zhyx12** [comment]
One more question, when training SSL model(with pseudo label), one argument is "--init-weights /path/to/inital_weights", should we train from pretrained DeepLab-V2 or finetune the segmentation model get from previous adaptation step(without pseudo label). Item 3 in Issue [16](https://github.com/liyunsheng13/BDL/issues/16#issue-484047566) use the word finetune.

**Comment by liyunsheng13** [comment]
You can use the pretrained DeepLab-V2. You can also train DeepLab-V2 with synthetic data first and then use it as initial model. The results make no difference

**Comment by wj-zhang** [comment]
> One more question, when training SSL model(with pseudo label), one argument is "--init-weights /path/to/inital_weights", should we train from pretrained DeepLab-V2 or finetune the segmentation model get from previous adaptation step(without pseudo label). Item 3 in Issue [16](https://github.com/liyunsheng13/BDL/issues/16#issue-484047566) use the word finetune.

Hi! Do you reproduce the best result in the paper? What initial weights do you use, is it pretrained DeepLab-V2?

**Comment by zhyx12** [comment]
I use the SSL_step2 model as initial weights which can achieve about 47.37 (mIoU). I test the mIoU every 2k iterations and the best result I can get is 48.3~48.5.

**Comment by wj-zhang** [comment]
> I use the SSL_step2 model as initial weights which can achieve about 47.37 (mIoU). I test the mIoU every 2k iterations and the best result I can get is 48.3~48.5.

Thank you very much for your reply! I used the DeepLab-V2 model as initial weights and obtained 48.17 at the last iteration (120000). I will try your setting and thanks again!
